source/frontend/src/features/mapSideBar/acquisition/common/GenerateForm/hooks/useGenerateH0443.ts

```diff
@@ -62,10 +55,65 @@ export const useGenerateH0443 = () => {
             : [],
         owner_names: owners.map<string>(x => getOwnerName(x)) || [],
         owner_contact: contactOwner !== undefined ? new Api_GenerateOwner(contactOwner) : null,
-        property_coordinator: new Api_GeneratePerson(coordinatorPerson),
-        property_agent: new Api_GeneratePerson(agentPerson),
+        property_coordinator: null,
+        property_agent: null,
       };
 
+      // Get the property coordinator by checking if it's a person or an org.
+      if (propertyCoordinator) {
+        if (propertyCoordinator.personId) {
+          const personConceptResponse = await getPersonConcept(propertyCoordinator?.personId);
+
+          h0443Data.property_coordinator = personConceptResponse?.data
+            ? new Api_GeneratePerson(personConceptResponse.data)
+            : null;
+        } else if (propertyCoordinator.organizationId) {
+          const organizationConceptResponse = await getOrganizationConcept(
+            propertyCoordinator?.organizationId,
+          );
+
+          if (
+            organizationConceptResponse &&
+            organizationConceptResponse?.data &&
+            organizationConceptResponse?.data.organizationPersons?.length &&
+            propertyCoordinator.primaryContactId
+          ) {
+            const personResponse = await getPersonConcept(propertyCoordinator.primaryContactId);
+
+            h0443Data.property_coordinator = personResponse.data
+              ? new Api_GeneratePerson(personResponse.data)
+              : null;
```

Comment:
nit: according to github/codecov there are no unit tests covering this new functionality. Please consider adding additional tests for that.

![image](https://github.com/bcgov/PSP/assets/24322224/4d5b9c81-906f-4184-928d-21dfe4963c08)

Reply:
Updated